core/ast.h

```diff
@@ -317,6 +318,15 @@ struct Index : public AST {
     { }
 };
 
+struct Slice : public AST {
```

Comment:
Is there an advantage to having this as a separate AST if it can only be used inside the [ ] of an Index AST?